src/connection_string.ts

```diff
@@ -785,26 +779,842 @@ function parseURI(uri: string): { srv: boolean; url: URL; hosts: string[] } {
+        if (rp) return rp;
+        else throw new MongoParseError(`Cannot make read preference from ${JSON.stringify(value)}`);
+      }
+      if (typeof value === 'string') {
+        const rpOpts = {
+          hedge: options.readPreference?.hedge,
+          maxStalenessSeconds: options.readPreference?.maxStalenessSeconds
+        };
+        return new ReadPreference(
+          value as ReadPreferenceMode,
+          options.readPreference?.tags,
+          rpOpts
+        );
+      }
+    }
+  },
+  readPreferenceTags: {
+    transform({ values }) {
+      // TODO!!!!!!!
+      const finalTags: TagSet = Object.create(null);
+      for (const tag of values) {
+        if (typeof tag === 'string') {
+          for (const [k, v] of Object.entries(toRecord(tag))) {
+            finalTags[k] = v;
+          }
+        }
+        if (isRecord(tag)) {
+          for (const [k, v] of Object.entries(tag)) {
+            finalTags[k] = v;
+          }
+        }
+      }
+    }
+  },
+  reconnectInterval: {
+    type: 'uint'
+  },
+  reconnectTries: {
+    type: 'uint'
+  },
+  replicaSet: {
+    type: 'string'
+  },
+  retryReads: {
+    type: 'boolean'
+  },
+  retryWrites: {
+    type: 'boolean'
+  },
+  serializeFunctions: {
+    type: 'boolean'
+  },
+  serverSelectionTimeoutMS: {
+    type: 'uint'
+  },
+  serverSelectionTryOnce: {
+    type: 'boolean'
+  },
+  servername: {
+    type: 'string'
+  },
+  socketTimeoutMS: {
+    type: 'uint'
+  },
+  ssl: {
+    rename: 'tls',
+    deprecated: true,
+    type: 'boolean'
+  },
+  sslCA: {
+    deprecated: true,
+    rename: 'ca',
+    type: 'asIs'
+  },
+  sslCRL: {
+    rename: 'crl',
+    type: 'asIs'
+  },
+  sslCert: {
+    deprecated: true,
+    rename: 'cert',
+    type: 'asIs'
+  },
+  sslKey: {
+    deprecated: true,
+    rename: 'key',
+    type: 'asIs'
+  },
+  sslPass: {
+    deprecated: true,
+    rename: 'passphrase',
+    type: 'string'
+  },
+  sslValidate: {
+    rename: 'rejectUnauthorized',
+    type: 'boolean'
+  },
+  tls: {
+    type: 'boolean'
+  },
+  tlsAllowInvalidCertificates: {
+    type: 'boolean'
+  },
+  tlsAllowInvalidHostnames: {
+    type: 'boolean'
+  },
+  tlsCAFile: {
+    rename: 'ca',
+    type: 'asIs'
+  },
+  tlsCertificateFile: {
+    rename: 'cert',
+    type: 'asIs'
+  },
+  tlsCertificateKeyFile: {
+    rename: 'key',
+    type: 'asIs'
+  },
+  tlsCertificateKeyFilePassword: {
+    rename: 'passphrase',
+    type: 'asIs'
+  },
+  tlsInsecure: {
+    type: 'boolean'
+  },
+  user: {
+    rename: 'credentials',
+    transform({ values: [value], options }) {
+      return new MongoCredentials({ ...options.credentials, username: String(value) });
+    }
+  } as OptionDescriptor,
+  username: {
+    rename: 'credentials',
+    transform({ values: [value], options }) {
+      return new MongoCredentials({ ...options.credentials, username: String(value) });
+    }
+  },
+  validateOptions: {
+    type: 'boolean'
+  },
```

Comment:
This appears to have been removed as well, I think there might be others here such as `waitQueueMultiple`